src/navigation/checkout/Restaurant.js

```diff
@@ -48,18 +55,50 @@ class Restaurant extends Component {
     })
   }
 
+  renderNotAvailableWarning(restaurant) {
+    if (!this.state.isAvailable) {
+        return (
+          <DangerAlert
+            text={`${this.props.t('RESTAURANT_CLOSED_AND_NOT_AVAILABLE', {
+              datetime: moment(restaurant.nextOpeningDate).calendar(moment(), {
+                sameElse: 'llll',
+              }),
+            })}`}
+          />
+        )
+    }
+  }
+
+  renderClosedNowWarning(restaurant) {
+    if (this.state.isAvailable && !restaurant.isOpen) {
+      return (
+        <DangerAlert
+          text={`${this.props.t('RESTAURANT_CLOSED_BUT_OPENS', {
+            datetime: moment(restaurant.nextOpeningDate).calendar(moment(), {
+              sameElse: 'llll',
+            }),
+          })}`}
+        />
+      )
+    }
+  }
+
   render() {
 
     const { navigate } = this.props.navigation
     const restaurant = this.props.route.params?.restaurant
     const { isCartEmpty, menu } = this.props
 
+    restaurant.isAvailable = this.state.isAvailable
```

Comment:
How can this work on a `const`? 🤔

Reply:
Because is an object. You can declare an object as a `const` and is possible to make changes on its properties/fields. What you can't do is re assign that variable.
https://stackoverflow.com/a/59516731